pkg/agent/validations.go

```diff
@@ -0,0 +1,88 @@
+	"sufficient-packet-loss-requirement-for-role":     "Packet loss",
+	"has-inventory":                                   "Hardware information",
+	"has-min-cpu-cores":                               "Minimum CPU cores",
+	"has-min-memory":                                  "Minimum Memory",
+	"has-min-valid-disks":                             "Minimum disks of required size",
+	"has-cpu-cores-for-role":                          "Minimum CPU cores for selected role",
+	"has-memory-for-role":                             "Minimum memory for selected role",
+	"hostname-unique":                                 "Unique hostname",
+	"hostname-valid":                                  "Valid hostname",
+	"connected":                                       "Connected",
+	"media-connected":                                 "Media Connected",
+	"machine-cidr-defined":                            "Machine CIDR",
+	"belongs-to-machine-cidr":                         "Belongs to machine CIDR",
+	"ignition-downloadable":                           "Ignition file downloadable",
+	"belongs-to-majority-group":                       "Belongs to majority connected group",
+	"valid-platform-network-settings":                 "Platform network settings",
+	"ntp-synced":                                      "NTP synchronization",
+	"container-images-available":                      "Container images availability",
+	"lso-requirements-satisfied":                      "LSO requirements",
+	"ocs-requirements-satisfied":                      "OCS requirements",
+	"sufficient-installation-disk-speed":              "Installation disk speed",
+	"cnv-requirements-satisfied":                      "CNV requirements",
+	"api-domain-name-resolved-correctly":              "API domain name resolution",
+	"api-int-domain-name-resolved-correctly":          "API internal domain name resolution",
+	"apps-domain-name-resolved-correctly":             "Application ingress domain name resolution",
+	"dns-wildcard-not-configured":                     "DNS wildcard not configured",
+	"non-overlapping-subnets":                         "Non overlapping subnets",
+	"vsphere-disk-uuid-enabled":                       "Vsphere disk uuidenabled",
+}
+
+func checkHostsValidations(cluster *models.Cluster, log *logrus.Logger) bool {
+
+	hostsValidationsOk := true
+
+	for _, h := range cluster.Hosts {
+		validationsInfo := common.ValidationsStatus{}
+		err := json.Unmarshal([]byte(h.ValidationsInfo), &validationsInfo)
+		if err != nil {
+			log.WithError(err).Error("Unable to verify cluster hosts validations")
+			return false
+		}
+
+		for category, validationResults := range validationsInfo {
+			for _, r := range validationResults {
+				switch r.Status {
+				case validationFailure, validationError:
+					hostsValidationsOk = false
+
+					label := r.ID
+					if v, ok := hostValidationLabels[r.ID]; ok {
+						label = v
+					}
+
+					log.WithFields(logrus.Fields{
```

Comment:
When there is a validation failure, the failure is printed out multiple times until the command times out. What do you think about keeping track of failures and only printing it out once?

Reply:
Since the validation failures could be fixed meanwhile by the user, the idea is to re-evaluate completely the current state at every loop: in this way the user could get a feedback if the fix(es) applied succeeded or not. I'm not sure that printing them only once could help under this point of view (maybe we should reduce the check frequency)

Reply:
I'll try anyhow to cache the latest messages and reprint only in case something changed, to see if it improves somehow